lib/include/public/PauseManager.hpp

```diff
@@ -0,0 +1,48 @@
+//
+// Copyright (c) 2015-2020 Microsoft Corporation and Contributors.
+// SPDX-License-Identifier: Apache-2.0
+//
+#ifndef PAUSEMANAGER_HPP
+#define PAUSEMANAGER_HPP
+
+#include <atomic>
+#include <chrono>
+#include <condition_variable>
+#include <mutex>
+
+#include "ctmacros.hpp"
+
+namespace MAT_NS_BEGIN
+{
+    class PauseManager {
+        public:
+
+        static void PauseActivity() noexcept;
+        static void ResumeActivity() noexcept;
+        static void QuiesceWait() noexcept;
+        static void QuiesceWaitFor(std::chrono::milliseconds timeout) noexcept;
```

Comment:
Is there a need for `QuiesceWaitFor`? I am trying to understand the use-case here. In UWP the apps usually ask for time-limited deferral. It could be something that is managed internally in PAL, i.e. reading the default deferral timeout value from `ILogConfiguration`. That way the defaults can be applied, and customers may be able to override their defaults - to give a hint to SDK how long it may wait for certain activities (such as how long to wait for flush to compete, or how long to wait for HTTP requests cancellation).